console/program/src/data_types/plaintext_type/mod.rs

```diff
@@ -26,14 +26,60 @@ pub enum PlaintextType<N: Network> {
     /// A literal type contains its type name.
     /// The format of the type is `<type_name>`.
     Literal(LiteralType),
-    /// An struct type contains its identifier.
+    /// A struct type contains its identifier.
     /// The format of the type is `<identifier>`.
     Struct(Identifier<N>),
+    /// An external struct type contains its locator.
+    /// The format of the type is `<program_id>/<identifier>`.
+    ExternalStruct(Locator<N>),
     /// An array type contains its element type and length.
     /// The format of the type is `[<element_type>; <length>]`.
     Array(ArrayType<N>),
 }
 
+impl<N: Network> PlaintextType<N> {
+    /// Are the two types equivalent for the purposes of static checking?
+    ///
+    /// Since struct types are compared by structure, we can't determine equality
+    /// by only looking at their names.
+    pub fn equal_or_structs(&self, rhs: &Self) -> bool {
+        use PlaintextType::*;
+
+        match (self, rhs) {
+            (ExternalStruct(..) | Struct(..), ExternalStruct(..) | Struct(..)) => true,
```

Comment:
What if we have two `Struct` with the same `Identifier`?

Reply:
They would not be equal right?

Reply:
Ah I read this pattern wrong. 
Follow on, do we want two structs with different `Identifier`s to be equal or not equal?

Reply:
Based on the description, if structs have the same data but different names, they are still the same.